cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/OracleGenerator.scala

```diff
@@ -0,0 +1,408 @@
+        val producedExprs = mutable.LinkedHashMap.newBuilder[String, FieldExpr]
+        val groupStrs = group.bys.map { by =>
+
+          val fieldExpr = exprMap(by.field.name)
+          val as = by.as.getOrElse(by.field)
+          val groupExpr = parseGroupByFunc(by, fieldExpr.refExpr)
+          val newExpr = s"$quote${as.name}$quote"
+
+
+          producedExprs += (as.name -> FieldExpr(newExpr, groupExpr))
+          s"$groupExpr"
+        }
+        val groupStr = s"group by ${groupStrs.mkString(",")}"
+        appendIfNotEmpty(queryBuilder, groupStr)
+        group.aggregates.foreach { aggr =>
+          val fieldExpr = exprMap(aggr.field.name)
+          val aggrExpr = parseAggregateFunc(aggr, fieldExpr.refExpr)
+          val newExpr = s"$quote${aggr.as.name}$quote"
+          producedExprs += aggr.as.name -> FieldExpr(newExpr, aggrExpr)
+        }
+
+        if (!group.lookups.isEmpty) {
+          val producedExprMap = producedExprs.result().toMap
+          val newExprMap =
+            producedExprMap.map {
+              case (field, expr) => field -> FieldExpr(s"$groupedLookupSourceVar.$quote$field$quote", s"$groupedLookupSourceVar.$quote$field$quote")
+            }
+          queryBuilder.insert(0, s"from (\n${parseProject(producedExprMap)}\n")
+          queryBuilder.append(s"\n) $groupedLookupSourceVar\n")
+          val resultAfterLookup = parseLookup(group.lookups, newExprMap, queryBuilder, true)
+          ParsedResult(Seq.empty, resultAfterLookup.exprMap)
+        } else {
+          ParsedResult(Seq.empty, producedExprs.result().toMap)
+        }
+
+      case None => ParsedResult(Seq(""), exprMap)
+    }
+  }
+
+  protected def parseCreate(create: CreateView, schemaMap: Map[String, AbstractSchema]): String = {
+
+    val (temporalSchemaMap, lookupSchemaMap) = GeneratorUtil.splitSchemaMap(schemaMap)
+    val sourceSchema = temporalSchemaMap(create.query.dataset)
+    val resultSchema = calcResultSchema(create.query, sourceSchema)
+    val ddl: String = genDDL(create.dataset, sourceSchema)
+    val primaryKeystr:Seq[String] = for (pk<- sourceSchema.primaryKey)yield{
+      pk.name
+    }
+    val measurementStrd:Seq[String] = for (d <- sourceSchema.measurement) yield {
+      "d.\""+d.name+"\""
+    }
+    val measurementStrs:Seq[String] = for (d <- sourceSchema.measurement) yield {
+      "s.\""+d.name+"\""
+    }
+    var dimensionStrd:Seq[String] = Seq()
+    for (d<- sourceSchema.dimension){
+      if (d.dataType != DataType.Hierarchy){
+        dimensionStrd= dimensionStrd :+"d.\"" + d.name + "\""
+      }
+    }
+    var dimensionStrs:Seq[String] = Seq()
+    for (d<- sourceSchema.dimension){
+      if (d.dataType == DataType.Hierarchy){
+      }
+      else {
+        dimensionStrs = dimensionStrs:+"s.\"" + d.name + "\""
+      }
+    }
+    var conflictsting = ""
+    for (pkname<- primaryKeystr){
+      conflictsting += "d.\""+pkname+"\" = s.\"" + pkname+"\" and "
+    }
+    val insert = s"""
```

Comment:
Is this whole construction of `insert` statement the same as previous `parseAppend` function? If so, can we reuse them by abstracting the code to a new function or using one of them.